### ISSUE TITLE
ValueError in .add_year.interpolate_2d() with pandas 1.4.0

### ISSUE THREAD

**Issue by LauWien** [iiasa/message_ix#559]
The tests `test_add_year.py::test_add_year_cli` and `test_add_year.py::test_add_year` start to fail seemingly randomly in various PRs on different OS e.g. [here](https://github.com/iiasa/message_ix/runs/5125036229?check_suite_focus=true). Looking deeper, failing test are using `pandas==1.4.0`, passing tests are using e.g. `pandas==1.1.5`. With upgrading pandas to 1.4.0 locally these errors can be replicated, while with version 1.1.5 the test succeed.

**Comment by khaeru** [comment]
The fundamental cause here is code that's not compatible with current pandas; the failing CI is only a symptom. The offending lines are: https://github.com/iiasa/message_ix/blob/1df3619a5223194171cc55197f65858ba62f16f2/message_ix/tools/add_year/__init__.py#L849-L851
```
ValueError: cannot set using a list-like indexer with a different length than the value
```

NB #494 was started to simplify these very long methods that mostly duplicate upstream features and are also quite slow, but I lacked clear tests for the expected behaviour for some combinations of arguments.

**Comment by LauWien** [comment]
> The fundamental cause here is code that's not compatible with current pandas;

Yes, was already checking where and how to adapt the code. Thank you for linking the PR!

**Comment by khaeru** [comment]
https://github.com/iiasa/message_ix/pull/557#issuecomment-1048604890
> It looks like there were several regressions related to `pd.DataFrame.loc` fixed in pandas 1.4.1: https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.4.1.html#fixed-regressions.

i.e. perhaps this is only with pandas 1.4.0 precisely.
